src/interface/blas1_interface.hpp

```diff
@@ -811,14 +811,36 @@ typename sb_handle_t::event_t _rotmg(
   auto d1_view = make_vector_view(_d1, inc, vector_size);
   auto d2_view = make_vector_view(_d2, inc, vector_size);
   auto x1_view = make_vector_view(_x1, inc, vector_size);
-  auto y1_view = make_vector_view(_y1, inc, vector_size);
   auto param_view = make_vector_view(_param, inc, param_size);
 
-  auto operation =
-      Rotmg<decltype(d1_view)>(d1_view, d2_view, x1_view, y1_view, param_view);
-  auto ret = sb_handle.execute(operation, _dependencies);
+  if constexpr (std::is_arithmetic_v<container_3_t>) {
+    constexpr helper::AllocType mem_type = std::is_pointer_v<container_0_t>
+                                               ? helper::AllocType::usm
+                                               : helper::AllocType::buffer;
+    auto _y1_tmp = blas::helper::allocate<mem_type, container_3_t>(
```

Comment:
Thanks for your PR, @s-Nick.

I think that  `_y1_tmp` needs to be deallocated when everything is done.

Reply:
Thank you for your review @pgorlani 

yes, you are right. I addressed it in [a4e833c](https://github.com/codeplaysoftware/portBLAS/pull/532/commits/a4e833cccf69255b43ce9c870677eeb208801305) and update it in [c128d67](https://github.com/codeplaysoftware/portBLAS/pull/532/commits/c128d670119e99598bf257afc5ebe9caedce28ae) to avoid `host_task` usage that causes some issues.